Added Spark 3.1 compatibility note. (#5673)
getsentry/sentry-docs: src/platforms/python/guides/pyspark/index.mdx

## Google Cloud Dataproc

> This integration can be set up to be used with [Google Cloud Dataproc](https://cloud.google.com/dataproc/). It is recommended that Cloud Dataproc image version 1.4 or 2.0 be used as they come with Spark 2.4 and 3.1, respectively (required by the worker integration).

Reviewer: ```suggestion
This integration can be set up for [Google Cloud Dataproc](https://cloud.google.com/dataproc/). It's recommended that Cloud Dataproc image version 1.4 or 2.0 be used with Spark 2.4 and 3.1, respectively, (as required by the worker integration).
```
